akka-http-core/src/main/scala/akka/http/impl/engine/http2/hpack/HeaderCompression.scala

```diff
@@ -44,7 +42,7 @@ private[http2] object HeaderCompression extends GraphStage[FlowShape[FrameEvent,
         case ParsedHeadersFrame(streamId, endStream, kvs, prioInfo) =>
           kvs.foreach {
             case (key, value) =>
-              encoder.encodeHeader(os, key.getBytes(HeaderDecompression.UTF8), value.getBytes(HeaderDecompression.UTF8), false)
+              encoder.encodeHeader(os, key, value.asInstanceOf[String], false)
```

Comment:
`case (key, value: String)` ?  (Also, why is this cast even needed, can the value be something else than a string here?)

Reply:
> case (key, value: String)

:1+:

>  (Also, why is this cast even needed, can the value be something else than a string here?)

Because we have only model in `Http2Substream` for both incoming and outgoing streams. The parsing optimization here requires that we can put `AnyRef` into the map. It isn't used like that on the rendering side.